process/sync/baseForkDetector.go

```diff
@@ -61,21 +62,17 @@ func (bfd *baseForkDetector) checkBlockBasicValidity(
 	nonceDif := int64(header.GetNonce()) - int64(bfd.finalCheckpoint().nonce)
 	//TODO: Analyze if the acceptance of some headers which came for the next round could generate some attack vectors
 	nextRound := bfd.rounder.Index() + 1
+	genesisTimeFromHeader := bfd.computeGenesisTimeFromHeader(header)
 
 	bfd.blackListHandler.Sweep()
 	if bfd.blackListHandler.Has(string(header.GetPrevHash())) {
-		//TODO: Should be done some tests to reconsider adding here to the black list also this received header,
-		// which is bound to a previous black listed header.
-		err := bfd.blackListHandler.Add(string(headerHash))
-		if err != nil {
-			log.Trace("add requested item with error",
-				"error", err.Error(),
-				"key", headerHash)
-
-		}
-
+		process.AddHeaderToBlackList(bfd.blackListHandler, headerHash)
 		return process.ErrHeaderIsBlackListed
 	}
+	if genesisTimeFromHeader != bfd.genesisTime {
```

Comment:
I think this check is better in the interceptor.

Reply:
added TODO to remove this check when implementation on interceptors side will be done